tests/child.cpp

```diff
@@ -105,7 +105,7 @@ TEST(childproc, child_exit_err)
   child->run();
   wait_for(child.get(), 1000);
   EXPECT_FALSE(child->is_alive());
-  EXPECT_EQ(child->exit_code(), 2);
+  EXPECT_TRUE(child->exit_code() > 0 && child->exit_code() <= 2);
```

Comment:
```suggestion
  EXPECT_TRUE(child->exit_code() > 0);
```

nit. As we only care about non zero exit we might as well only check for that. In case some other distro has a diff return value too